simple_graph.py

```diff
@@ -4,6 +4,8 @@
 http://www.martinbroadhurst.com/graph-data-structures.html
 """
 
+from Queue import Queue
```

Comment:
This is actually not the queue you want to use.  This queue is meant to be used in multi-threaded apps.  From the module docs:

> The Queue module implements multi-producer, multi-consumer queues. It is especially useful in threaded programming when information must be exchanged safely between multiple threads.

Instead, you want to look at the `deque` class from the `collections` module.  It implements an efficient data structure that can be used both as a queue and a stack.